### DISCUSSION TITLE
IP wireless camera?

### DISCUSSION THREAD

**Discussion by babsenfred** [aaronwmorris/indi-allsky#1514] [Q&A]
Hello, trying to understand the INDI camera protocol... Is it possible to have a wireless (wifi) camera outside and let it send al the pictures to a computer running Linux (e.g. Ubuntu) that is inside to do all the image processing? And if yes, how do I know for sure that a camera will work? Something specific there has to be in the specifications?
Thanks!

**Comment by aaronwmorris** [comment]
Yes, this is possible.

If you have a wireless network IP camera, indi-allsky can use the pyCurl Camera to download images from the camera if it supports image download from its web server.  Picking a camera can be difficult as it must support downloading images via the web server.  (RTSP streams are not supported).  I do not really maintain a list of cameras that support web based downloads.  I have seen cameras from Reolink and Amcrest cameras work with indi-allsky.

This is an example of an Amcrest camera:  https://www.youtube.com/watch?v=wOo3-mLgGOk

**Reply by morrowwm** [reply]
I wrote a python script to convert an RTSP stream to periodically output a jpeg file, also copying it to a network location. That might be a starting point?

https://github.com/morrowwm/cameraImage

It adds timestamp text to each image, and some other processing which could be stripped out.

I'm using this to build a daily time lapse video from a Tapo CS320WS security camera. It does a decent job of adjusting exposure for night. It is decent resolution (2560 × 1440) and 100° field of view. I'm not using this for allsky, just monitoring our yard.

**Reply by morrowwm** [reply]
I suspect you could likely do it with a periodic  ffmpeg call, too.
